### PR TITLE
Add support for `target_index_key` config.

### PR DESCRIPTION
**Pull request by ashb**
This is one of the PRs to come out of uken/fluent-plugin-elasticsearch#152 "Time.parse
is slow - allow other methods of finding index name"

### CODE REVIEW

**Comment by ashb** [comment]
@pitr Here's the first of the PRs -- this one was easiest as I had the code already. I'll code up the other one tomorrow.

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -151,22 +152,26 @@ def write(chunk)
...
-          record['@timestamp'] = record[@time_key]
-        else
-          record.merge!({"@timestamp" => Time.at(time).to_datetime.to_s})
-        end
-        if @utc_index
-          target_index = "#{@logstash_prefix}-#{Time.at(time).getutc.strftime("#{@logstash_dateformat}")}"
+      if @target_index_key && record[@target_index_key]
+        target_index = record.delete @target_index_key
+      else
+        if @logstash_format
```
Comment (pitr) [inline lib/fluent/plugin/out_elasticsearch.rb]: can this be `elsif`?

File: README.md [hunk 2]
```diff
@@ -154,6 +155,39 @@ utc_index true
...
+  "@target_index": "logstash-2014.12.19"
+}
+```
+
+The output would be
+
+```
+{
+  "title": "developer",
+  "@timstamp": "2014-12-19T08:01:03Z",
```
Comment (pitr) [inline README.md]: should be `timEstamp`

Comment (ashb) [inline README.md]: Good spot!

File: lib/fluent/plugin/out_elasticsearch.rb (hunk 1 above)
Comment (ashb) [inline lib/fluent/plugin/out_elasticsearch.rb]: Oh yes should be fine -- I'm not quite sure why I wrote it that way.

**Comment by ashb** [comment]
@pitr updated PR to address your comments.

**Pull request merged**